### PR TITLE
Added PowerVS Cluster Create & Destroy

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
**What this PR does / why we need it**:

Added implementation for PowerVS Cluster Create and destroy 

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/fixtures/example.go
Comment (mkumatag) [inline api/fixtures/example.go:6]: move this in the below section of k8s imports

File: api/v1alpha1/hostedcluster_types.go
Comment (mkumatag) [inline api/v1alpha1/hostedcluster_types.go:580]: I don't see this is used anywhere in this change, can we remove this if not

File: api/v1alpha1/hostedcluster_types.go
Comment (mkumatag) [inline api/v1alpha1/hostedcluster_types.go:585]: I don't see this is used anywhere in this change, can we remove this if not

File: cmd/cluster/powervs/create.go
Comment (mkumatag) [inline cmd/cluster/powervs/create.go:48]: As the subcommand is already powervs, can we just remove the `powervs-` prefix from these options and also in the struct.

File: cmd/install/assets/assets.go
Comment (mkumatag) [inline cmd/install/assets/assets.go:42]: thes crds aren't related to powervs, let revert back the changes

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (mkumatag) [inline docs/content/how-to/powervs/create-cluster-powervs.md:19]: can we remove is this field isn't mandatory?

File: docs/content/how-to/powervs/prerequisites-and-env-guide.md
Comment (mkumatag) [inline docs/content/how-to/powervs/prerequisites-and-env-guide.md:4]: good to add link to the IBM Cloud CIS service here

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/fixtures/example.go
Comment (dharaneeshvrd) [reply api/fixtures/example.go]: moved

File: api/v1alpha1/hostedcluster_types.go
Comment (dharaneeshvrd) [reply api/v1alpha1/hostedcluster_types.go]: Removed

File: cmd/cluster/powervs/create.go
Comment (dharaneeshvrd) [reply cmd/cluster/powervs/create.go]: Removed powervs- prefix in flags as well as in structs

File: cmd/install/assets/assets.go
Comment (dharaneeshvrd) [reply cmd/install/assets/assets.go]: Reverted back

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md]: removed

File: docs/content/how-to/powervs/prerequisites-and-env-guide.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/prerequisites-and-env-guide.md]: added the link!

File: cmd/install/assets/assets.go
Comment (dharaneeshvrd) [reply cmd/install/assets/assets.go:39]: line is reordered, will rectify it.

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (hardys) [inline docs/content/how-to/powervs/create-cluster-powervs.md:23]: I think this should be moved to the `Create cluster with infra created separately` section below, and should include `--output-file` and `--infra-id` ?

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (hardys) [inline docs/content/how-to/powervs/create-cluster-powervs.md:30]: If this section omits the `create infra` step above this is not needed, but should be included in the `Create cluster with infra created separately` example - also you have a typo in both places s/exmaple/example

File: docs/content/how-to/powervs/create-infra-powervs-separately.md
Comment (hardys) [inline docs/content/how-to/powervs/create-infra-powervs-separately.md:37]: typo here

File: api/fixtures/example.go
Comment (enxebre) [inline api/fixtures/example.go:433]: I'd expect this to be granular perms for each component?

File: api/fixtures/example.go
Comment (enxebre) [inline api/fixtures/example.go:459]: any reason to not use `getIngressServicePublishingStrategyMapping(o.NetworkType)`?

File: api/fixtures/example.go
Comment (mkumatag) [reply api/fixtures/example.go:433]: Yes, this is something we need to explore as a overall approach how we want to address for the PowerVS flow, may be reuse the some of the work we have done for the [CCO](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md#ibmcloud), wondering if can merge this code for now adding TODO here and start working on that once we have all the code merged for the PowerVS?

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md:23]: This should be a example for create cluster command only, fixed now.

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md]: Fixed the commands in respective place, please take a look.
Fixed the typo.

File: docs/content/how-to/powervs/create-infra-powervs-separately.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-infra-powervs-separately.md]: Fixed

File: cmd/cluster/powervs/destroy.go
Comment (dharaneeshvrd) [reply cmd/cluster/powervs/destroy.go:59]: Yes, this should be --infra-json, modified.

File: api/fixtures/example.go
Comment (dharaneeshvrd) [reply api/fixtures/example.go]: We can use `getIngressServicePublishingStrategyMapping()` func itself. Modified the code to use this func.

File: cmd/cluster/powervs/create.go
Comment (enxebre) [inline cmd/cluster/powervs/create.go:49]: Can we clarify in the text output these are for the default NodePool created by the cli? e.g "System type used to host the machines in the NodePool" (e.g: s922, e980, e880)"

**Comment by enxebre** [comment]
just [one nit](https://github.com/openshift/hypershift/pull/1381#discussion_r879231530) lgtm otherwise, PTAL @hardys

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (hardys) [inline docs/content/how-to/powervs/create-cluster-powervs.md:51]: I think these three lines should be moved to the following section, and we should add the `create infra` CLI step so it's clear how to generate the `OUTPUT_INFRA_FILE`?

**Comment by hardys** [comment]
> just [one nit](https://github.com/openshift/hypershift/pull/1381#discussion_r879231530) lgtm otherwise, PTAL @hardys

A couple more comments but nothing to block on, overall this lgtm :)

File: cmd/cluster/powervs/create.go
Comment (dharaneeshvrd) [reply cmd/cluster/powervs/create.go]: Added default values

File: cmd/cluster/powervs/create.go
Comment (dharaneeshvrd) [reply cmd/cluster/powervs/create.go]: Modified, ptal.

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md]: removed the \ at the end

File: docs/content/how-to/powervs/create-cluster-powervs.md
Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md:51]: In previous line I have mentioned, 
`You can create infra separately and use it to create Hypershift cluster which reduces the infra creation time.`
I need to give a link to the section where infra creation separately described. Also wanted to give extra info for the user on `OUTPUT_INFRA_FILE` since that is required for the following command mentioned in 55.

Comment (hardys) [reply docs/content/how-to/powervs/create-cluster-powervs.md:51]: Ack, my main point was that everything related to creating infra separately should go into the `Create cluster with infra created separately` section below :)

Comment (dharaneeshvrd) [reply docs/content/how-to/powervs/create-cluster-powervs.md]: Makes sense, moved that part to the `create-infra-powervs-separately.md` page itself. 
Ptal. Will merge the commits if it is ok.

**Comment by dharaneeshvrd** [comment]
@hardys @enxebre I think I have addressed all the comments, please let me know if you have any more comments.

**Comment by enxebre** [comment]
/approve
cc @hardys @alvaroaleman @sjenning @csrwng for another pair of eyes to tag and merge.

**Pull request merged**
